### DISCUSSION TITLE
Help with navigating long lists of completion candidates

### DISCUSSION THREAD

**Discussion by max6166** [emacs-helm/helm#2654] [General]
I'm new to helm and still exploring how it works. I've searched but haven't found any info regarding an issue I often come across. The problem occurs under many circumstances, but I'll use the following as an example.

I organize many of my filenames using a subject and then subtopic. This is a made-up example, but let's say I have the following files:

- media-books-fiction.org
- media-books-fiction-scifi.org
- media-books-nonfiction-science.org
- media-books-nonfiction-history.org
- etc.

If I want to org-refile a heading in org-mode, helm will present me with a very long list of candidates. The candidates consist of an alphabetical list of matching filenames followed by all the headings and subheadings within each file. 

So if I have 150 headings and subheadings in `- media-books-fiction.org`, those are the first 150 completion candidates, followed by the same for each file that matches.

So if I don't remember which 1st level headings I want, for example, I have to scroll through hundreds of 2nd and 3rd level headings to see all the 1st level headings.

Or if I just want to see a list of all the files beginning with the word "media" so I can see what options are available refile targets, I have to scroll through the same entire list of hundreds subheadings.

My assumption is that these issues are due to my lack of knowledge as  a helm newbie, so any guidance would be greatly appreciated.

**Comment by thierryvolpiatto** [comment]
max6166 ***@***.***> writes:

> 1.  ( ) text/plain          (*) text/html           
>
> I'm new to helm and still exploring how it works. I've searched but haven't found any info regarding an issue I often come across. The problem occurs under many circumstances, but I'll use
> the following as an example.
>
> I organize many of my filenames using a subject and then subtopic. This is a made-up example, but let's say I have the following files:
>
>   • media-books-fiction.org
>   • media-books-fiction-scifi.org
>   • media-books-nonfiction-science.org
>   • media-books-nonfiction-history.org
>   • etc.
>
> If I want to org-refile a heading in org-mode, helm will present me
> with a very long list of candidates. The candidates consist of an
> alphabetical list of matching filenames followed by all the headings
> and subheadings within each file.

My org usage is very limited, when I do org-refile with cursor on a
heading in a org buffer, I have a list of headings of current buffer, no
file names.  This is probably because org-refile-targets is nil (the
default), what are the settings you use for this (The docstring is
wrong so I am unable to set this properly) ?


>
>
> So if I have 150 headings and subheadings in - media-books-fiction.org, those are the first 150 completion candidates, followed by the same for each file that matches.
>
> So if I don't remember which 1st level headings I want, for example, I have to scroll through hundreds of 2nd and 3rd level headings to see all the 1st level headings.
>
> Or if I just want to see a list of all the files beginning with the word "media" so I can see what options are available refile targets, I have to scroll through the same entire list of
> hundreds subheadings.
>
> My assumption is that these issues are due to my lack of knowledge as a helm newbie, so any guidance would be greatly appreciated.
>
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you are subscribed to this thread.*Message ID: ***@***.***>

-- 
Thierry

**Reply by max6166** [reply]
> My org usage is very limited, when I do org-refile with cursor on a
heading in a org buffer, I have a list of headings of current buffer, no
file names.  This is probably because org-refile-targets is nil (the
default), what are the settings you use for this (The docstring is
wrong so I am unable to set this properly) ?

Yes, I should have mentioned that `org-refile-targets` to `((org-agenda-files :maxlevel . 8))` in my configuration. This means that all my agenda files are refile targets up to 8 levels deep. Now that I see it, I'm going to reduce the header levels to 5 though.

For me, this worked well before I started using helm because of incremental completion. In other words, I could slowly navigate the completion options level by level, rather than having them all in one huge list.

**Comment by thierryvolpiatto** [comment]
max6166 ***@***.***> writes:

> 1.  ( ) text/plain          (*) text/html           
>
>     My org usage is very limited, when I do org-refile with cursor on a
>     heading in a org buffer, I have a list of headings of current buffer, no
>     file names. This is probably because org-refile-targets is nil (the
>     default), what are the settings you use for this (The docstring is
>     wrong so I am unable to set this properly) ?
>    
> Yes, I should have mentioned that org-refile-targets to
> ((org-agenda-files :maxlevel . 8)) in my configuration. This means
> that all my agenda files are refile targets up to 8 levels deep.  Now
> that I see it, I'm going to reduce the header levels to 5 though.

Ok, thanks.

> For me, this worked well before I started using helm because of
> incremental completion. In other words, I could slowly navigate the
> completion options level by level, rather than having them all in one
> huge list.

So you should have a list of headers, why don't you narrow down the list
by typing e.g. "media" ?

> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you commented.*Message ID: ***@***.***>

-- 
Thierry

**Comment by thierryvolpiatto** [comment]
Thierry Volpiatto ***@***.***> writes:

> max6166 ***@***.***> writes:
>
>> 1.  ( ) text/plain          (*) text/html           
>>
>>     My org usage is very limited, when I do org-refile with cursor on a
>>     heading in a org buffer, I have a list of headings of current buffer, no
>>     file names. This is probably because org-refile-targets is nil (the
>>     default), what are the settings you use for this (The docstring is
>>     wrong so I am unable to set this properly) ?
>>    
>> Yes, I should have mentioned that org-refile-targets to
>> ((org-agenda-files :maxlevel . 8)) in my configuration. This means
>> that all my agenda files are refile targets up to 8 levels deep.  Now
>> that I see it, I'm going to reduce the header levels to 5 though.
>
> Ok, thanks.
>
>> For me, this worked well before I started using helm because of
>> incremental completion. In other words, I could slowly navigate the
>> completion options level by level, rather than having them all in one
>> huge list.
>
> So you should have a list of headers, why don't you narrow down the list
> by typing e.g. "media" ?

You can also refile easily from the helm-org package.

>> —
>> Reply to this email directly, view it on GitHub, or unsubscribe.
>> You are receiving this because you commented.*Message ID: ***@***.***>

-- 
Thierry
